src/Stratis.Bitcoin.Features.RPC/RPCMiddleware.cs

```diff
@@ -125,6 +126,12 @@ private async Task HandleRpcInvokeExceptionAsync(HttpContext httpContext, Except
                 httpContext.Response.ContentType = ContentType;
                 await httpContext.Response.WriteAsync(response.ToString(Formatting.Indented));
             }
+            else if (ex is BlockNotFoundException)
```

Comment:
Do we *have* to throw an exception in the middleware? Can't we return some message?

Reply:
That's the way error handling is currently architected in our middleware. Didn't want to introduce a new mechanism for it as it would require a fairly big rewrite of the RPC Middleware.